control-plane/piSerializer.cpp

```diff
@@ -0,0 +1,1471 @@
+/*
```

Comment:
I don't understand whether the goal is to generate a PI which is compatible with the way BMv2 does things today, or this is meant to be a generic PI generation tool. 

How about we sit together for a couple of hours and do by hand the PI generation for a simple P4-16 program? I think that except extern objects everything can be done in a very generic way. Extern objects are trickier, because they are architecture-dependent. Maybe this code can be broken down into architecture-specific and architecture-independent pieces.